scanpy/readwrite.py

```diff
@@ -277,15 +277,49 @@ def _read_v3_10x_h5(filename, *, start=None):
                 (data, dsets['indices'], dsets['indptr']),
                 shape=(N, M),
             )
+            obs_dict = dict(obs_names=dsets['barcodes'].astype(str))
+            var_dict = dict(
+                var_names=dsets['name'].astype(str),
+            )
+            if 'gene_id' not in dsets:
+                var_dict.update(
+                    {
+                        'gene_ids': dsets['id'].astype(str),
+                        'feature_types': dsets['feature_type'].astype(str),
```

Comment:
Factor out this line, which is common to both branches of the if/else.

Reply:
Fixed